finances.js

```diff
@@ -14,24 +14,92 @@ let bankAccount = {
 
 function main() {
   console.log("Welcome to your personal finance manager");
+
   while (true) {
     console.log(
       "What do you want to do?\n" +
-      "[b] show latest balance\n" +
-      "[bd]show balance for a specific date\n" +
-      "[t] show all transactions\n" +
-      "[a] add a transaction\n" +
-      "[d] delete a transaction\n" +
-      "[s] show statistics\n" +
-      "[q] quit"
+        "[b] show latest balance\n" +
+        "[bd]show balance for a specific date\n" +
+        "[t] show all transactions\n" +
+        "[a] add a transaction\n" +
+        "[d] delete a transaction\n" +
+        "[s] show statistics\n" +
+        "[q] quit"
     );
     let choice = prompt();
     if (choice === "q") {
       break;
     } else if (choice === "b") {
       showCurrentBalance();
+    } else if (choice === "bd") {
+      showBalanceForSpecificDate();
+    } else if (choice === "t") {
+      showAllTransactions();
+    } else if (choice === "a") {
+      addATransaction();
+    } else if (choice === "d") {
+      deleteATransaction();
+    }
+  }
+}
+
+function showCurrentBalance() {
+  let sum = bankAccount.startBalance;
+  for (let allTransactions of bankAccount.transactions) {
+    sum += allTransactions.amount;
+  }
+  console.log(`Current balance: ${sum}`);
+}
+
+function showBalanceForSpecificDate() {
+  console.log("Please insert the date (e.g 2022-09-31) ");
+  let balanceOfSpecificDate = 0;
+  let userInputDate = prompt();
+  for (let allTransactions of bankAccount.transactions) {
```

Comment:
Hier hätte ich gemeint, dass der Kontostand zu einem bestimmten Datum dadurch berechnet werden kann, dass man mit dem Anfangskontostand beginnt und dann alle Transaktionen bis zum genannten Datum dazu addiert.

Die restlichen, weiter in der Zukunft liegenden Transaktionen werden nicht berücksichtigt.

Reply:
Achso okay, danke, das habe ich wohl falsch verstanden. Werde ich gleich umändern.